pytensor/gradient.py

```diff
@@ -128,9 +128,6 @@ def fiter_variable(self, other):
             " a symbolic placeholder."
         )
 
-    def may_share_memory(a, b):
```

Comment:
Removing this avoids having to check for aliasing in the first place, during the function call